Feature/issue873

mdmiller53 opened the pull request:
code to take a ParticipantBarcode list and create the metadata tables based on it.  the log is 
[top_processing.txt](https://github.com/isb-cgc/ISB-CGC-data-proc/files/50071/top_processing.txt)
@medullaskyline please code review

medullaskyline commented:
These may be naive comments...

1) In [this line](https://github.com/isb-cgc/ISB-CGC-data-proc/pull/1/files#diff-c296250ef7059191a99aa1ba30476f05R79) the [[]] looks like it's the params keyword argument in the update class method. a) when calling a keyword argument don't you have to name it? Like this: `ISBCGC_database_helper.update(config, table_create_stmt, log, params=[[]])`? b) Not sure why it's `[[]]` instead of just `[]`. c) since the params kwarg defaults to `[]` anyway, could you just leave this blank and have this: `ISBCGC_database_helper.update(config, table_create_stmt, log)`?

2) This line probably works but it makes me want to test it to see if it gives something like 'barcode1, barcode2, barcode3' instead of "'barcode1', 'barcode2','barcode3'" which is what you want I think.

I didn't review the generateProtectedFileInfo function because it isn't used.

mdmiller53 commented:
added comments.

@medullaskyline 
added comments
for 1a) see [stackoverflow](http://stackoverflow.com/questions/1419046/python-normal-arguments-vs-keyword-arguments) for a pretty good description but if you have default arguments but specify values for all the other arguments on the list, as long as the arg value is in the right place, it doesn't need its name.
1b) if you look at the method being called, it loops through the params input arg.  in this case there are no parameters for the SQL so it needs to be represented by an empty list in the params, thus the nested list.  another usage is to use parameter markers and send multiple paramsets in params--this is actually more efficient since the sql plan is stashed and doesn't have to be redone.   but since there were ~600 barcodes, this avoided having to deal with the last set which wouled need a different number of parameter markers.  oh, and if params were just '[]', nothing happens! it needs the empty list to work (so the answer to 1c)
2) the quotes are added in the loop above when the barcodes are chunked into lists (see log which shows that they get quoted properly).  for proper sql, one needs 'barcode1', 'barcode2','barcode3' in the in clause
